pennylane/devices/experimental/device_api.py

```diff
@@ -0,0 +1,555 @@
+
+            """
+            return res
+
+        circuit_batch = [circuits] if isinstance(circuits, QuantumTape) else circuits
+        return circuit_batch, blank_postprocessing_fn
+
+    @abc.abstractmethod
+    def execute(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ):
+        """Execute a circuit or a batch of circuits and turn it into results.
+
+        Args:
+            circuits (Union[QuantumTape, Sequence[QuantumTape]]): the quantum circuits to be executed
+            execution_config (ExecutionConfig): a datastructure with additional information required for execution
+
+        Returns:
+            TensorLike, tuple[TensorLike], tuple[tuple[TensorLike]]: A numeric result of the computation.
+
+        **Interface parameters:**
+
+        Note that the parameters contained within the quantum script may contain interface-specific data types, such as
+        ``torch.Tensor`` or ``jax.Array``. If the device does not wish to handle interface-specific parameters, they
+        can implement an optional "internal preprocessing" step that converts all parameters to vanilla numpy. A convenience
+        transform implementing this will be provided. This step allows device to be transparent to things like jitting if they
+        so choose.
+
+        .. details::
+            :title: Return Shape
+
+            The result for each :class:`~.QuantumTape` must match the shape specified by :class:`~.QuantumTape.shape`.
+
+            The level of priority for dimensions from outer dimension to inner dimension is:
+
+            1. Quantum Script in batch
+            2. Shot choice in a shot vector
+            3. Measurement in the quantum script
+            4. Parameter broadcasting
+            5. Measurement shape for array-valued measurements like probabilities
+
+            For a batch of quantum scripts with multiple measurements, a shot vector, and parameter broadcasting:
+
+            * ``result[0]``: the results for the first script
+            * ``result[0][0]``: the first shot number in the shot vector
+            * ``result[0][0][0]``: the first measurement in the quantum script
+            * ``result[0][0][0][0]``: the first parameter broadcasting choice
+            * ``result[0][0][0][0][0]``: the first value for an array-valued measurement
+
+            With the exception of quantum script batches, dimensions with only a single component should be eliminated.
+
+            For example:
+
+            With a single script and a single measurement process, execute should return just the
+            measurement value in a numpy array. ``shape`` currently accepts a device, as historically devices
+            stored shot information. In the future, this method will accept an ``ExecutionConfig`` instead.
+
+            >>> qs = qml.tape.QuantumTape(measurements=qml.expval(qml.PauliZ(0))])
```

Comment:
kinda sneaky variable naming 😈 I don't mind subtly bringing scripts into discussion, but I do kinda worry that a small variable ending in "s" sounds like a plural?

Reply:
```suggestion
            >>> tape = qml.tape.QuantumTape(measurements=qml.expval(qml.PauliZ(0))])
```